druid/examples/styled_text.rs

```diff
@@ -57,26 +59,39 @@ fn ui_builder() -> impl Widget<AppData> {
     });
 
     // This is druid's default text style.
-    // It's set by theme::LABEL_COLOR and theme::TEXT_SIZE_NORMAL
+    // It's set by theme::LABEL_COLOR, theme::TEXT_SIZE_NORMAL, and theme::FONT_NAME
     let label =
         Label::new(|data: &String, _env: &_| format!("Default: {}", data)).lens(AppData::text);
 
-    // The text_color and text_size builder methods can override the defaults
-    // provided by the theme by passing in a Key or a concrete value.
+    // The text_color, text_size, and font builder methods can override the
+    // defaults provided by the theme by passing in a Key or a concrete value.
     //
-    // In this example, text_color receives a Key from the theme, while
-    // text_size gets a custom key which we set with the env_scope wrapper.
-    let styled_label =
-        Label::new(|data: &AppData, _env: &_| format!("Size {:.1}: {}", data.size, data.text))
-            .with_text_color(theme::PRIMARY_LIGHT)
-            .with_text_size(MY_CUSTOM_TEXT_SIZE)
-            .background(my_painter)
-            .on_click(|_, data, _| {
-                data.size *= 1.1;
-            })
-            .env_scope(|env: &mut druid::Env, data: &AppData| {
-                env.set(MY_CUSTOM_TEXT_SIZE, data.size);
-            });
+    // In this example, text_color receives a Key from the theme, text_size
+    // gets a custom key which we set with the env_scope wrapper, and the
+    // default font key (theme::FONT_NAME) is overridden in the env_scope
+    // wrapper. (Like text_color and text_size, the font can be set using the
+    // with_font builder method, but overriding here makes it easy to fall back
+    // to the default font)
+    let styled_label = Label::new(|data: &AppData, _env: &_| {
+        format!(
+            "Size {:.1}{}: {}",
+            data.size,
+            if data.mono { " mono" } else { "" },
+            data.text
+        )
+    })
+    .with_text_color(theme::PRIMARY_LIGHT)
+    .with_text_size(MY_CUSTOM_TEXT_SIZE)
+    .background(my_painter)
+    .on_click(|_, data, _| {
+        data.size *= 1.1;
+    })
+    .env_scope(|env: &mut druid::Env, data: &AppData| {
+        env.set(MY_CUSTOM_TEXT_SIZE, data.size);
+        if data.mono {
+            env.set(theme::FONT_NAME, "monospace");
```

Comment:
The current thinking on "idiomatic" druid (a WIP) is to use a custom key and override that (like with MY_CUSTOM_TEXT_SIZE) instead of overriding a default key

Reply:
Will do.

Reply:
Trying to set the custom font name to the current theme name causes errors due to borrowing the `env` when I borrow the font name. I think adjustments may need to be made to the current env system to make this work.

Reply:
you might need to clone the name before passing it back in or something?

Curious to see what your problem was.